app/controllers/posts_controller.rb

```diff
@@ -9,4 +9,27 @@ def show
     @post = @user.posts.find(params[:id])
     @comments = @post.comments
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.author_id = current_user.id
+    @post.comments_counter = 0
+    @post.likes_counter = 0
```

Comment:
[OPTIONAL]
Please try to keep the controllers thinner by keeping the not relevant logic outside of the controllers. 👍